drivers/soundwire/slave.c

```diff
@@ -53,6 +52,9 @@ int sdw_slave_add(struct sdw_bus *bus,
 	slave->dev.of_node = of_node_get(to_of_node(fwnode));
 	slave->dev.type = &sdw_slave_type;
 	slave->dev.groups = sdw_slave_status_attr_groups;
+	/* To make sure fwnode is a primary fwnode */
+	fwnode->secondary = NULL;
+	set_primary_fwnode(&slave->dev, fwnode);
```

Comment:
The commit message needs to be updated, e.g.

Properties may be added to a codec as a secondary fwnode. When the slave device is removed, the fwnode remains tagged as secondary, which causes a side effect if the slave device is re-added in module load/unload tests. Forcing the fwnode as primary is required, otherwise the dev->fwnode will be overridden with the properties.